srv_hijacker/srv_hijacker.py

```diff
@@ -59,7 +60,55 @@ def patched_f(host, port, family=0, type=0, proto=0, flags=0):
     return patched_f
 
 
-def hijack(host_regex, srv_dns_host=None, srv_dns_port=None):
+class PatchError(BaseException):
+    pass
+
+
+def _patch_psycopg2(host_regex, srv_resolver):
+    try:
+        import psycopg2
+        from psycopg2.extensions import parse_dsn, make_dsn
+    except ImportError as e:
+        raise PatchError(f"failed to import psycopg2. {e}")
+
+    def patch_psycopg2_connect(fn):
+        @wraps(fn)
+        def wrapper(dsn, connection_factory=None, *args, **kwargs):
+            config = parse_dsn(dsn)
+            host = config.get("host")
+            if host is None:
+                host = kwargs.get("host")
+
+            if host is None:
+                # Host may be left out to use localhost or
+                # possibly set using environment variables, nothing
+                # we can do in either case.
+                logger.error("'host' parameter is not present in call to psycopg2.connect, "
+                             "DNS resolution might not work properly")
+
+                return fn(dsn, connection_factory, *args, **kwargs)
+
+            if re.search(host_regex, host):
+                logger.debug("Host %s matched SRV regex, resolving", host)
+                host, port = resolve_srv_record(host, srv_resolver)
+                config["host"] = host
+                config["port"] = port
+
+            dsn = make_dsn(**config)
+
+            return fn(dsn, connection_factory, *args, **kwargs)
+
+        return wrapper
+
+    psycopg2._connect = patch_psycopg2_connect(psycopg2._connect)
+
+
+SUPPORTED_LIBS = {
+    "psycopg2": _patch_psycopg2,
+}
+
+
+def hijack(host_regex, srv_dns_host=None, srv_dns_port=None, patch_everything=True, raise_error=False, **libs):
```

Comment:
Would it be okay to replace `raise_error` & `patch_everything` with a `libraries_to_patch` option that accepts an array of strings? That way the user explicitly mentions what they want patched, and if it doesn't work out we always throw a `PatchError`.

Reply:
And let's drop the `**libs` option, take an explicit keyword argument instead - better for long-term API maintainability (if we decide to rename the field, or change the meaning, this makes it easier to do)

Reply:
So basically, my suggestion here is - drop `raise_error`, `path_everything` and `libs`. Only accept `libraries_to_patch`, and always raise an error if patch fails.